game/src/js/scenes/TitleScene.js

```diff
@@ -41,15 +40,38 @@ export default class TitleScene extends BaseScene {
     });
 
     //controls, credits, offline mode buttons
-    let settingsButton = new Button(this, 450, 350, 'SETTINGS');
-    settingsButton.buttonDown(() => {
-      new DOMModal(this, 'settings', {
-        cancelButtonSelector: '.exit',
-        onCancel: (modal) => {
-          modal.close();
-        }
+    if(ServiceWorker.isSupported()) {
+      const assets = this.cache.json.get('assets');
+      let offlineMode = localStorage.getItem('offlineStatus');
+      offlineMode = (offlineMode == 'true')
+      if (offlineMode == null) {
+        offlineMode = false;
+      }
+      let serviceWorker = new ServiceWorker();
+      let settingsButton = new Button(this, 450, 350, 'SETTINGS', serviceWorker.isRegistered());
+      settingsButton.buttonDown(() => {
+        new DOMModal(this, 'settings', {
+          cancelButtonSelector: '.exit',
+          acceptButtonSelector: '#settingsCheck',
+          onCancel: (modal) => {
+            localStorage.setItem('offlineStatus' , offlineMode)
+            modal.close();
+          },
+          onAccept: (modal) => {
+            offlineMode = modal.modal.querySelector('#settingsCheck').checked;        
+            if (offlineMode == true) {
+              serviceWorker.register().then(function() {
+                serviceWorker.cacheAssets(assets);
+              })
+            }
+            else {
+              serviceWorker.unregister();
+            }
+          },
+          data: {swCheck: offlineMode}
```

Comment:
`swCheck: serviceWorker.isRegistered()`

I'd rename `swCheck` to something a bit more descriptive like `offlineMode`.